test/puppetlabs/puppetdb/testutils/log.clj

```diff
@@ -0,0 +1,173 @@
+(ns puppetlabs.puppetdb.testutils.log
+  (:require
+   [puppetlabs.kitchensink.core :as kitchensink]
+   [puppetlabs.puppetdb.testutils :refer [temp-file]]
+   [me.raynes.fs :as fs])
+  (:import
+   [ch.qos.logback.core Appender FileAppender]
+   [ch.qos.logback.core.filter EvaluatorFilter]
+   [ch.qos.logback.core.spi FilterReply LifeCycle]
+   [ch.qos.logback.classic Level Logger]
+   [ch.qos.logback.classic.boolex JaninoEventEvaluator]
+   [ch.qos.logback.classic.encoder PatternLayoutEncoder]
+   [org.slf4j LoggerFactory]))
+
+(defn- find-logger [id]
+  (.getLogger (LoggerFactory/getILoggerFactory)
+              (if (class? id) id (str id))))
+
+(defmacro with-log-level
+  "Sets the (logback) log level for the logger specified by logger-id
+  during the execution of body.  If logger-id is not a class, it is
+  converted via str, and the level must be a clojure.tools.logging
+  key, i.e. :info, :error, etc."
+  [logger-id level & body]
+  ;; Specify the root logger via org.slf4j.Logger/ROOT_LOGGER_NAME.
+  ;; Assumes use of logback (i.e. logger supports Levels).
+  `(let [logger# (#'puppetlabs.puppetdb.testutils.log/find-logger ~logger-id)
+         original-level# (.getLevel logger#)]
+     (.setLevel logger# (case ~level
```

Comment:
Hmm, yeah, not sure - I normally don't initiate the try until successfully establishing the thing I want to undo in the finally, but given the semantics here, it would probably work fine to have the initial setLevel inside the try - as long as setLevel is sane, it shouldn't matter, but in this case it shouldn't hurt and might be safer to have it in the try...